test/apps/uct_info/Makefile.in

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (C) 2022, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+
+all: uct_info uct_info_static
+
+clean:
+	rm -rf uct_info uct_info_static
+
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+PKG_CONFIG_CMD = PKG_CONFIG_PATH=@libdir@/pkgconfig:$(PKG_CONFIG_PATH) pkg-config
+
+UCX_CFLAGS = $(CFLAGS) $(shell $(PKG_CONFIG_CMD) ucx-uct --cflags)
+
+UCX_LDFLAGS = $(LDFLAGS) $(shell $(PKG_CONFIG_CMD) ucx-uct --libs)
+
+UCX_STATIC_LDFLAGS = -static $(shell $(PKG_CONFIG_CMD) --libs --static @uct_static_modules@ ucx-uct) -pthread -lrt -ldl
+
+uct_info: @abs_srcdir@/uct_info.c
+	$(CC) -o $@ $? $(UCX_CFLAGS) $(UCX_LDFLAGS) -Wl,-rpath=@libdir@
+
+uct_info_static: @abs_srcdir@/uct_info.c
+	$(CC) -o $@ $? $(UCX_CFLAGS) $(UCX_STATIC_LDFLAGS)
```

Comment:
IMO it's weird that makefile in ucx depends on installed pkg-config.
Since we want to test end-user scenario here, so the test build process of uct_info_static should be (well documented) SH script in buildlib. We would share this script with users as demo for static build.

Reply:
> IMO it's weird that makefile in ucx depends on installed pkg-config.

simplified makefile to make it independent from installed configuration.
used .in template to workaround Automake issue.